### ISSUE TITLE
Does not work anymore with old Windows versions (<1809)

### ISSUE THREAD

**Issue by erentar** [mintty/wsltty#206]
I recently downgraded from 1903 to 1709 and thus downgrade from wsl2 to wsl. Wsltty does no longer work, i have uninstalled deleted all local files etc. and reinstalled multiple times and still cant get it to work. How can i solve this issue?

**Comment by Biswa96** [comment]
Use the old wsltty 3.0.2.3.

**Comment by erentar** [comment]
I've installed and tried all versions from 3.0.6 to 3.0.2 and still cant get it to work. The window pops up for 1 frame and disappears. The contents say something about wslbridge not working, i cant get more info. I've tried to record the screen and go frame by frame but it did not get captured. If you can tell me how to funnel the screen contents out i can put it here

**Comment by erentar** [comment]
Mostly backwards compatibility, the keyboard language i was using got completely revamped and sends different keycodes now. I'll try to upgrade to 1809 as i know that update does not break the keyboard.

**Comment by erentar** [comment]
Nevermind, managed to get it working. Installing 3.0.2.3 and overwriting wslbridge with [2017 ](https://github.com/rprichard/wslbridge/releases/tag/0.2.4)version works.

I would upgrade to 1809 but thanks to Micocock™, i cant. they removed the update installer.

**Comment by llinfeng** [comment]
On enterprise-owned machines (in my cases, University-owned computers), the Windows OS build can be as old as 1709 as of 2019-12-03. Then, downloading [Version 3.0.2.3](https://github.com/mintty/wsltty/releases/tag/3.0.2.3) is the way to go.

I only found it necessary to run the [installer for 3.0.2.3](https://github.com/mintty/wsltty/releases/tag/3.0.2.3) for one time on the "old" Windows computers. The installer shall completely revive the `wsltty` executables on these computers. I did not find it necessary to **overwrite wslbridge** as mentioned in @erentar's comment.

**Comment by mintty** [comment]
> Windows OS build can be as old as 1709

If there is strong request for support of older Windows versions, wslbridge2 could be enabled with two patches:
* optionally use distro-guid for invocation (I've successfully tried that locally already)
* optionally supply a backend path in Windows format already, to avoid using wslpath

@Biswa96, what do you think? I may take care about both patches.

**Comment by llinfeng** [comment]
TL;DR: a really minor point on rendering the icons.

I find certain icons were "squeezed" on the "old Windows machines" running [wsltty Version 3.0.2.3](https://github.com/mintty/wsltty/releases/tag/3.0.2.3). I copied and pasted the entire `c:/users/llinfeng/AppData/Roaming/wsltty` directory to the old Windows machine (Build 1709), and `zsh` is showing an odd-looking command prompt, as below:
![image](https://user-images.githubusercontent.com/5890407/70192392-a9217080-16c9-11ea-8a16-9c5256072d39.png)
* The `➜` symbol is super narrow (copied directly from `wsltty` console);
* The ` ✗` symbol is also super unnoticeable as well.


I have reproduced the same display problem on another work-computer, which also runs Windows 1709 and is likely to be updated in 2021 or so 👍. [I use Pengwin for WSL, and have downloaded from Windows Store directly. On "newer Windows machines" of mine, there was no such display issue.]

**Comment by llinfeng** [comment]
Also, a funny screen capture, where I toggled the font-size back and forth. ` ✗` had a better luck to be displayed normally, for a certain font-size. 
![Icon and Symbol](https://user-images.githubusercontent.com/5890407/70193016-a7f14300-16cb-11ea-8adf-a397c673e192.gif)


The ` ✗` with proper width is obtained under the following setting.
![image](https://user-images.githubusercontent.com/5890407/70193000-96a83680-16cb-11ea-8069-860abc9100e4.png)

**Comment by mintty** [comment]
This is really a different issue; anyway, you're talking about an older version. 3.0.6 had changes about symbol rendering.

**Comment by llinfeng** [comment]
Agreed. May I use newer versions of `wsltty` on old Windows Builds (1709, for example)? Say, to specify an older version of _wslbridge_ for the up-to-date `wsltty` to use, so that the terminal won't flash-quit on the old Windows machines?

**Comment by mintty** [comment]
The internal interface changed with the transition to wslbridge2, so you cannot really mix.
To support older Windows versions, the above changes to wslbridge2, and according adapations in mintty would be needed.

**Comment by mintty** [comment]
Actually, this refers to the implicit WSL invocation using mintty option `--WSL`. If you don't use that, you'd be free to involve any bridge you like, for example `mintty /bin/winpty wsl`.
Note, however, that winpty limits access to mintty features, that's in fact why wslbridge was once created.
You could also `mintty /bin/ssh ...` to WSL if it's running already and you find out its ssh server address...

**Comment by Biswa96** [comment]
Is this still an issue in 2020? According to [Microsoft Windows 10 release information][1], version 1809 will be out of service in next month.

[1]: https://docs.microsoft.com/en-us/windows/release-information/

**Comment by mintty** [comment]
Thanks for this link. Actually the issue is against versions _before_ 1809, which are already out of service now according to that list, so I'll close the issue although it obviously still affects people who have not yet updated.
